tests/python/gaia-ui-tests/gaiatest/tests/functional/lockscreen/test_lockscreen_time_check.py

```diff
@@ -0,0 +1,45 @@
+import time
+
+from marionette.wait import Wait
+
+from gaiatest import GaiaTestCase
+from gaiatest.apps.lockscreen.app import LockScreen
+
+
+class TestLockScreen(GaiaTestCase):
+
+    _seconds_since_epoch = 1357043430  # the time is 12:30 GMT
+
+    def setUp(self):
+        GaiaTestCase.setUp(self)
+        #self.connect_to_local_area_network()
+
+    def test_lockscreen_time_check(self):
+
+        # disable time/timezone automatic update
+        self.data_layer.set_setting('time.clock.automatic-update.enabled', False)
+        self.data_layer.set_setting('time.timezone.automatic-update.enabled', False)
+
+        # set time manually to 12:30
+        self.data_layer.set_time(self._seconds_since_epoch * 1000)
+        self.data_layer.set_setting('time.timezone', 'Atlantic/Reykjavik')
+
+        self.device.lock()
+        lock_screen = LockScreen(self.marionette)
+
+        #check the time on the lockscreen and unlock.  Below check gives 10 minutes of windows
+        #on desktop b2g, it usually takes 8 minutes to get to this point
+        self.assertTrue('12:3' in lock_screen.time)
+        lock_screen.switch_to_frame()
+        lock_screen.unlock()
+
+        # set the time to be set automatically
+        self.data_layer.set_setting('time.clock.automatic-update.enabled', True)
+        self.data_layer.set_setting('time.timezone.automatic-update.enabled', True)
+        self.device.lock()
+
+        #wait until device is off and turn back on
+        Wait(self.marionette, timeout=20).until(
+            lambda m: self.device.is_screen_enabled == False)
+        self.device.turn_screen_on()
+        self.assertFalse('12:3' in lock_screen.time)
```

Comment:
it would fail if the new time is actually 12:3x.  but if this test is run on some really odd time, there is a chance that this test will fail.  I am thinking of ways to read the current time, then change it.  However, right now Bug 1122503 is blocking me from submitting this test.  (found it during the script writing process)